core/fixel/helpers.h

```diff
@@ -35,6 +35,19 @@ namespace MR
         : Exception(previous_exception, msg) {}
   };
 
+  namespace Peaks
+  {
+    FORCE_INLINE void check (const Header& in)
+    {
+      if (!in.datatype().is_floating_point())
+        throw Exception ("Image \"" + in.name() + "\" is not a valid peaks image: Does not contain floating-point data");
+      if (in.ndim() != 4)
```

Comment:
Might be a candidate for the new [check_effective_dimensionality()](https://github.com/MRtrix3/mrtrix3/blob/dev/core/image_helpers.h#L257-L268)...?

Reply:
4ffd3ae

Reply:
:+1: